### PR TITLE
Fix VCRPY security leak issue

### PR DESCRIPTION
**Pull request by sfc-gh-yuwang**
remove credential in cassettes files

### CODE REVIEW

File: src/snowflake/vcrpy/_constant.py [hunk 1]
```diff
@@ -4,13 +4,25 @@
 
 SNOWFLAKE_CREDENTIAL_HEADER_FIELDS = [
     "Authorization",
+    "x-amz-server-side-encryption-customer-key-MD5",
+    "x-amz-server-side-encryption-customer-key-md5",
```
Comment (sfc-gh-aling) [inline src/snowflake/vcrpy/_constant.py:8]: are the header fields case-sensitive -- `MD5` vs `md5`?

File: src/snowflake/vcrpy/_constant.py [hunk 2]
```diff
@@ -4,13 +4,25 @@
...
+    "x-amz-server-side-encryption-customer-key",
+    "x-amz-server-side-encryption-customer-algorithm",
+    "x-amz-id-2",
```
Comment (sfc-gh-aling) [inline src/snowflake/vcrpy/_constant.py:11]: is there a `x-amz-id-1`?

File: src/snowflake/vcrpy/snowflake_vcrpy_pytest_plugin.py [hunk 3]
```diff
@@ -28,13 +28,22 @@ def _process_request_recording(request):
...
         # scrub snowflake account information
         if request.host.endswith(".snowflakecomputing.com"):
             account = request.host.split(".snowflakecomputing.com")[0]
             request.uri = request.uri.replace(account, VOID_STRING)
 
+        if request.host.endswith(".amazonaws.com"):
+            account = request.host.split(".amazonaws.com")[0]
+            code = request.uri.split("/results/")[1].split("/")[0]
+            request.uri = request.uri.replace(account, VOID_STRING)
+            request.uri = request.uri.replace(code, VOID_STRING)
```
Comment (sfc-gh-aling) [inline src/snowflake/vcrpy/snowflake_vcrpy_pytest_plugin.py]: what about gcp and azure?

File: src/snowflake/vcrpy/snowflake_vcrpy_pytest_plugin.py [hunk 4]
```diff
@@ -28,13 +28,22 @@ def _process_request_recording(request):
                 key in SNOWFLAKE_REQUEST_ID_STRINGS
                 or key in SNOWFLAKE_DB_RELATED_FIELDS_IN_QUERY
             ):
-                request.uri = request.uri.replace(value, VOID_STRING)
+                if request.uri.find(value) ==-1:
+                    request.uri = request.uri.replace(urllib.parse.quote_plus(value), VOID_STRING)
+                else:
+                    request.uri = request.uri.replace(value, VOID_STRING)
```
Comment (sfc-gh-aling) [inline src/snowflake/vcrpy/snowflake_vcrpy_pytest_plugin.py]: nit:
```suggestion
                request.uri = request.uri.replace(urllib.parse.quote_plus(value) if request.uri.find(value) == -1 else value, VOID_STRING)
```

File: src/snowflake/vcrpy/_constant.py (hunk 2 above)
Comment (sfc-gh-yuwang) [reply src/snowflake/vcrpy/_constant.py:11]: no, there is only x-amz-id-0

File: src/snowflake/vcrpy/_constant.py (hunk 1 above)
Comment (sfc-gh-yuwang) [reply src/snowflake/vcrpy/_constant.py:8]: Yes, I think it is case sensitive

File: src/snowflake/vcrpy/snowflake_vcrpy_pytest_plugin.py (hunk 3 above)
Comment (sfc-gh-aling) [reply src/snowflake/vcrpy/snowflake_vcrpy_pytest_plugin.py]: let's create a JIRA and add a comment in the code that azure and gcp will be handled separately

File: src/snowflake/vcrpy/snowflake_vcrpy_pytest_plugin.py [hunk 5]
```diff
@@ -28,13 +28,18 @@ def _process_request_recording(request):
...
-
+                request.uri = request.uri.replace(urllib.parse.quote_plus(value) if request.uri.find(value) == -1 else value, VOID_STRING)
         # scrub snowflake account information
         if request.host.endswith(".snowflakecomputing.com"):
             account = request.host.split(".snowflakecomputing.com")[0]
             request.uri = request.uri.replace(account, VOID_STRING)
 
+        if request.host.endswith(".amazonaws.com"):
+            account = request.host.split(".amazonaws.com")[0]
+            code = request.uri.split("/results/")[1].split("/")[0]
```
Comment (sfc-gh-aling) [inline src/snowflake/vcrpy/snowflake_vcrpy_pytest_plugin.py:41]: can you also add a comment what this `code` is trying to extract?

**Review by sfc-gh-aling — APPROVED** [review]

**Pull request merged**